internal/app/machined/pkg/controllers/block/symlinks.go

```diff
@@ -0,0 +1,199 @@
+// This Source Code Form is subject to the terms of the Mozilla Public
+// License, v. 2.0. If a copy of the MPL was not distributed with this
+// file, You can obtain one at http://mozilla.org/MPL/2.0/.
+
+package block
+
+import (
+	"context"
+	"errors"
+	"fmt"
+	"io/fs"
+	"os"
+	"path/filepath"
+	"slices"
+
+	"github.com/cosi-project/runtime/pkg/controller"
+	"github.com/cosi-project/runtime/pkg/safe"
+	"github.com/cosi-project/runtime/pkg/state"
+	"github.com/siderolabs/gen/optional"
+	"go.uber.org/zap"
+	"golang.org/x/sys/unix"
+
+	"github.com/siderolabs/talos/internal/app/machined/pkg/controllers/block/internal/inotify"
+	"github.com/siderolabs/talos/pkg/machinery/resources/block"
+	"github.com/siderolabs/talos/pkg/machinery/resources/v1alpha1"
+)
+
+// SymlinksController provides a view of symlinks created by udevd to the blockdevices.
+type SymlinksController struct{}
+
+// Name implements controller.Controller interface.
+func (ctrl *SymlinksController) Name() string {
+	return "block.SymlinksController"
+}
+
+// Inputs implements controller.Controller interface.
+func (ctrl *SymlinksController) Inputs() []controller.Input {
+	return []controller.Input{
+		{
+			Namespace: v1alpha1.NamespaceName,
+			Type:      v1alpha1.ServiceType,
+			ID:        optional.Some("udevd"),
+			Kind:      controller.InputWeak,
+		},
+	}
+}
+
+// Outputs implements controller.Controller interface.
+func (ctrl *SymlinksController) Outputs() []controller.Output {
+	return []controller.Output{
+		{
+			Type: block.SymlinkType,
+			Kind: controller.OutputExclusive,
+		},
+	}
+}
+
+const baseDevDiskPath = "/dev/disk"
+
+var byDiskPaths = []string{
```

Comment:
any reference for these, better to add as a comment

Reply:
good point, I'm not sure if all of them exist always, they seem to be coming from udev rules in the end, and they are loosely defined

Reply:
I think I will rework them a bit to support better handling

Reply:
reworked this completely, now we don't hardcode the list